core/src/main/java/org/apache/iceberg/io/ContentCache.java

```diff
@@ -32,16 +32,15 @@
 import org.apache.iceberg.exceptions.NotFoundException;
 import org.apache.iceberg.exceptions.ValidationException;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
-import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Class that provides file-content caching during reading.
  *
- * <p>The file-content caching is initiated by calling {@link ContentCache#tryCache(FileIO, String,
- * long)}. Given a FileIO, a file location string, and file length that is within allowed limit,
+ * <p>The file-content caching is initiated by calling {@link ContentCache#tryCache(InputFile)}.
```

Comment:
The changes here aren't strictly necessary, but clean up several problems I noticed while updating the code to call `newInputFile(ManifestFile)` instead of `newInputFile(String,long)`. The class leaked the `CacheEntry` class in its public API (those methods are now deprecated) and had duplicate checks, overly verbose debug logs, and odd error handling. I've simplified it a bit and made the naming more clear for internal methods and classes.